### PR TITLE
DRAFT - Do not review

### PR DESCRIPTION
**Pull request by mch2**
Extract getTranslogDeletionPolicy method so it can be overriden by subclasses and provide custom DeletionPolicies.

Signed-off-by: Marc Handalian <handalm@amazon.com>

### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/index/translog/TranslogReader.java [hunk 1]
```diff
@@ -129,8 +129,9 @@ public int totalOperations() {
         return totalOperations;
     }
 
+    // we can provide an isBetween method here if we don't want to change visibility on this method.
     @Override
-    final Checkpoint getCheckpoint() {
+    public final Checkpoint getCheckpoint() {
```
Comment (mch2) [inline server/src/main/java/org/opensearch/index/translog/TranslogReader.java:134]: not totally in love with this, feel like it was package private for a reason.  I think an isBetween method here may be more appropriate.

**Comment by mch2** [comment]
Example of how this would be used here https://github.com/opensearch-project/cross-cluster-replication/pull/145

File: server/src/main/java/org/opensearch/index/translog/TranslogReader.java (hunk 1 above)
Comment (nknize) [reply server/src/main/java/org/opensearch/index/translog/TranslogReader.java:134]: Can't we keep it package private and have the plugin implement this in the same package (e.g., `o.opensearch.index.translog.ReplicationTranslogDeletionPolicy`) only exposing the public facing APIs needed for the plugin specific consumer classes?

Comment (mch2) [reply server/src/main/java/org/opensearch/index/translog/TranslogReader.java:134]: ah yep, done.

**Pull request closed without merging**